### ISSUE TITLE
Increased Android Project Size after Integrating Rive SDK

### ISSUE THREAD

**Issue by MuhHarisPK** [rive-app/rive-android#298]
I am experiencing an issue with my Android project where the project size has increased significantly after integrating the Rive SDK (formerly known as Flare). Prior to integrating the Rive SDK, my project size was within an acceptable range, but after adding the SDK, the size has increased beyond expectations.

**Comment by avadhbsd** [comment]
Thanks for posting this here Haris, can you please add the numbers as well so it can be confirmed if the increase in size is expected or if it indicates any other issues.

**Comment by MuhHarisPK** [comment]
yes @avadhbsd , Increased APK Size up to **12MB** from Current Size after Integrating Rive SDK

**Comment by umberto-sonnino** [comment]
Hi @MuhHarisPK, our library is ~10.8MB in its latest iteration, but that takes into account multiple architectures which should get stripped out by Google Play when downloading the APK directly from the Play Store. 
We currently support 4 different architectures, so on average the Rive library should add ~2.5MB to your final APK.

Please provide a few more details here:
- Did you build for debug or release? 
- Which version are you currently using? 
- Could you confirm if this is a Play Store APK or one that you built locally?

**Comment by MuhHarisPK** [comment]
Hi @umberto-sonnino 
-> We've built the release using version 7.0.0. Is this the latest version?

Regarding another issue: I can't seem to locate the animationView.getTextRunValue("name") method for handling dynamic text in this version 7.0.0.

**Comment by umberto-sonnino** [comment]
> Hi @umberto-sonnino -> We've built the release using version 7.0.0. Is this the latest version?
> 
> Regarding another issue: I can't seem to locate the animationView.getTextRunValue("name") method for handling dynamic text in this version 7.0.0.

The latest version is `8.2.2`([Sonatype link](https://central.sonatype.com/artifact/app.rive/rive-android))
Anything blocking you from upgrading to that? Some Text APIs aren't available until v8

**Comment by MuhHarisPK** [comment]
no blocker, please update in the doc. that's why we facing the confusion about latest version.
is version 8.2.2 stable? is there any critical crash on this version?

**Comment by umberto-sonnino** [comment]
> @umberto-sonnino no blocker, please update the doc. that's why we facing confusion about the latest version. is version 8.2.2 stable? Is there any critical crash on this version?

You can find the latest version of the library either on [Sonatype](https://central.sonatype.com/artifact/app.rive/rive-android) or via the [GitHub repo tags](https://github.com/rive-app/rive-android/tags), the latest stable version is the one at the top and currently it's `8.2.3`.

**Comment by PhilippNowak96** [comment]
Just dropping some comment here because we also came across this recently: APK Analyzer of Android Studio tells the following numbers in a release build APK:

- x86: 5.9 MB
- x86_64: 5.9 MB
- arm64-v8a: 5.3 MB
- armeabi-v7a: 4.4 MB

We are currently using version 8.5.0.

Are these numbers expected? Since they seem to be twice as high as they should be in total (20 MB instead of 10 MB)

**Comment by rcgonzalezf** [comment]
This is still an issue with the latest SDK, it will be great if Rive devs can squeeze more bytes since it impacts consumers of the library.
Just adding an idea here, that a potential solution could be to break the SDK in multiple pieces so consumers can only use what they need reducing the overall impact even with App Bundles, and release variants.
